src/helm/common/object_spec.py

```diff
@@ -13,7 +13,7 @@ class ObjectSpec:
     args: Dict[str, Any]
 
     def __hash__(self):
-        return hash((self.class_name, tuple((k, self.args[k]) for k in sorted(self.args.keys()))))
+        return hash((self.class_name, tuple((k, self.args[k]) if not isinstance(self.args[k], list) else (k, ",".join(self.args[k])) for k in sorted(self.args.keys()))))
```

Comment:
Two requests for you:

1. `",".join(self.args[k])` is not appropriate for hashing because it causes `["a", "b,c"]` to collide with `["a,b", "c"]`. Instead, you should just do `tuple(self.args[k])` which converts the list to a tuple, which is hashable.
2. Could you break this code into multiple lines? This is a long line with deep nesting of brackets, which makes this hard to read.